spec/feature/login_spec.rb

```diff
@@ -190,32 +190,92 @@
       )
     end
 
-    before do
-      expect(user.ephemeral_states.last).to be_nil
+    context "when the feature_flag_enforce_levels_of_authentication is not 'enabled'" do
+      before do
+        expect(user.ephemeral_states.last).to be_nil
+        log_in_from_app(application)
+      end
 
-      visit authorization_endpoint_url(client: application, scope: "openid")
-      user_is_returned_to_login_screen
+      it "records that the user has logged in with level-of-authentication 1" do
+        enter_mfa
 
-      fill_in "email", with: user.email
-      fill_in "password", with: user.password
-      click_on I18n.t("devise.sessions.new.fields.submit.label")
-    end
+        expect(user.reload.ephemeral_states.last&.level_of_authentication).to eq("level1")
+      end
 
-    it "records that the user has logged in with level-of-authentication 1" do
-      enter_mfa
+      context "when the user doesn't have MFA set up" do
+        let(:user) { FactoryBot.create(:user, :without_mfa) }
 
-      expect(user.reload.ephemeral_states.last&.level_of_authentication).to eq("level1")
+        it "records that the user has logged in with level-of-authentication 0" do
+          expect(user.reload.ephemeral_states.last&.level_of_authentication).to eq("level0")
+        end
+      end
     end
 
-    context "when the user doesn't have MFA set up" do
-      let(:user) { FactoryBot.create(:user, :without_mfa) }
+    context "when the feature_flag_enforce_levels_of_authentication is 'enabled'" do
+      context "for a level0 authorised user" do
+        let(:user) { FactoryBot.create(:user, :without_mfa) }
+
+        let(:level0_application) do
```

Comment:
There's no need to have a separate `level0_application` and `level1_application`-  just give the `application` (line 184) the `level0` and `level1` scopes

Reply:
Yep, I've seen that route now in the registration from oauth app spec.
Will transfer these and switch it out of draft in a bit.